File: AgentUp.Server/Features/Workspaces/DTOs/Workspace.cs
namespace AgentUp.Server.Features.Workspaces.DTOs;

public class Workspace
{
    public required string Id { get; init; }
    public required string DisplayName { get; init; }
    public required string RepositoryPath { get; init; }
    public required string WorktreePath { get; init; }
    public required string Branch { get; init; }
    public required string Commit { get; init; }
    public WorkspaceState State { get; set; } = WorkspaceState.Stopped;
    public IReadOnlyList<ApplicationInstance> Applications { get; init; } = [];
}

File: AgentUp.Mobile/src/features/workspaces/models/Workspace.ts
export type WorkspaceApplication = {
  name: string;
  state: string;
  kind?: 'Process' | 'Desktop';
  allocatedPorts?: Array<{
    allocatedPort: number;
    protocol: string;
  }>;
  capabilityStatus?: {
    capabilityId: string;
    requiredVersion?: string | null;
    canRun: boolean;
    messages: string[];
  };
};

export type Workspace = {
  id: string;
  displayName: string;
  repositoryPath: string;
  worktreePath: string;
  branch: string;
  commit: string;
  state: string;
  healthState?: string;
  applications?: WorkspaceApplication[];
};

export type CloneSourceRequest = {
  repository: string;
  branch: string;
};
